core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2953,6 +2979,166 @@ private void scheduleAsyncPersistenceInternal(LockedInodePath inodePath) throws
     inodePath.getInode().setPersistenceState(PersistenceState.TO_BE_PERSISTED);
   }
 
+  /**
+   * Syncs the Alluxio metadata with the UFS metadata, for a given path.
+   *
+   * @param path the root of the subtree to sync
+   * @param options the options to use for the method
+   * @return true if the sync was performed
+   */
+  private boolean syncMetadata(AlluxioURI path, SyncMetadataOptions options)
+      throws AccessControlException, FileDoesNotExistException, InvalidPathException {
+    if (!mUfsSyncPathCache
+        .shouldSyncPath(path.getPath(), options.getCommonOptions().getSyncIntervalMs())) {
+      return false;
+    }
+
+    // True if ufs metadata should be loaded later.
+    boolean loadMetadata = false;
+
+    DeleteOptions syncDeleteOptions =
+        DeleteOptions.defaults().setRecursive(true).setAlluxioOnly(true).setUnchecked(true);
+
+    // The high-level process for the syncing is:
+    // 1. Find all Alluxio paths which are not consistent with the corresponding UFS path.
+    //    This means the UFS path does not exist, or is different from the Alluxio metadata.
+    // 2. Delete those Alluxio paths which are not consistent with Alluxio. After this step, all
+    //    the paths in Alluxio are consistent with UFS, and there may be additional UFS paths to
+    //    load.
+    // 3. Load metadata from UFS.
+
+    try (JournalContext journalContext = createJournalContext();
+         LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE);
+         FileSystemMasterAuditContext auditContext =
+             createAuditContext("syncMetadata", path, null, inodePath.getInodeOrNull())) {
```

Comment:
does `loadMetadata` needs its own audit context, or can it share the callers? same question for `journalContext` and `LockedInodePath`

Reply:
I restructured this, to re-use the existing contexts.